instruments/keithley/keithley580.py

```diff
@@ -336,11 +336,12 @@ def get_status_word(self):
         """
         tries = 5
         statusword = ''
-        while statusword[:3] != '580' and tries != 0:
+        while statusword[:3] != b'580' and tries != 0:
             tries -= 1
             self.sendcmd('U0X')
             time.sleep(1)
-            statusword = self.query('')
+            self.sendcmd('')
```

Comment:
`self.query('')` sends a ':' which we now need to do by sending an empty command prior to using `read_raw`

Reply:
But `query` also strips the last character returned from the statusword. An additional character is stripped here, because the original author does not want to parse the termination character, which is sent to. So this right now breaks the original functionality!
Need to add a termination character to the statusword (as per manual) that is created in the test.

Reply:
This should be fixed now. Whatever the status word is, we only want to unpack 13 bytes form [3:16]. This restores all the changes to act in the same behavior as before. If the termination character is sent, `read_raw` will split it there and the complete status word will be [0:16] in size. If the termination character something different, then the status word will be [0:17] in length, however, the part of interest will still be included in [3:16]. This should thus take care of all cases.

Reply:
[Manual, see page 71](http://www.mhzelectronics.com/ebay/manuals/keithley_580_meter_manual.pdf): 
When the status word is read, does it send the terminator + `(CR)(LF)` or is the terminator equal to `(CR)(LF)` from the data format described above. The current implmentation will work unless the terminator + `(CR)(LF)` are sent and the terminator is equal to `(CR)(LF)`, i.e., send end of transmission twice. My feeling is that the latter wouldn't make much sense... (but then again)

Reply:
This is why I hate non-SCPI equipment

Reply:
> This is why I hate non-SCPI equipment

Very true... So I did some digging, PR #46 had a bug fix for stripping the terminator from parsing. The encoding string for the status word went wrong back then ('@8c2s2s2c' was changed to '@8c2s2s2'), which I actually corrected for now.

This means that the status word that is sent does contain one termination character at the end, so the manual is correct, just that it says the same things in two very different styles between the data format and the status word. 

For the test this would mean that I should create the status word without a byte for a random terminator, since a terminator is sent by `expected_protocol`. That would also mean that I can remove the `[3:16]` when unpacking the status word, since a valid status word should terminate there by itself. What do you think?